File: examples/databases/db-subscribe-params.swift
import JsBaoClient

// Bind values to a parameterized subscription — the bound `params.*` are
// visible to the subscription's `access` and `filter` CEL.
func watchTeamTickets(
  client: JsBaoClient,
  databaseId: String,
  handleChange: @escaping (DatabaseChangeEvent) -> Void
) throws -> () -> Void {
  // #region example
  let unsub = try client.databases.subscribe(
    databaseId: databaseId,
    subscriptionKey: "tickets-by-team",
    options: DatabaseSubscribeOptions(
      params: ["teamId": "eng"],
      onChange: { event in
        for change in event.changes { handleChange(change) }
      }
    )
  )
  // #endregion example
  return unsub
}

File: examples/databases/db-subscribe-params.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Bind values to a parameterized subscription — the bound `params.*` are
// visible to the subscription's `access` and `filter` CEL.
export function watchTeamTickets(
  client: JsBaoClient,
  databaseId: string,
  handleChange: (change: unknown) => void
) {
  // #region example
  const unsub = client.databases.subscribe(databaseId, "tickets-by-team", {
    params: { teamId: "eng" },
    onChange: (event) => {
      for (const change of event.changes) handleChange(change);
    },
  });
  // #endregion example
  return unsub;
}
